File: gerenciador de tarefas/models/Item.php
<?php

class Item {
    private $nome;
    private $concluido;
	private $prioridade;

    public function __construct($nome) { 
        $this->nome = $nome;
        $this->concluido = false;
    }
    
	public function getNome() {
		return $this->nome;
	}
	
	public function setNome($nome): self {
		$this->nome = $nome;
		return $this;
	}

	public function getConcluido() {
		return $this->concluido;
	}
	
	public function setConcluido($concluido): self {
		$this->concluido = $concluido;
		return $this;
	}
}

?>

File: java desktop/src/main/java/com/taniele/java_desktop/model/Item.java
package com.taniele.java_desktop.model;

public class Item implements Comparable<Item>{
    private String nome;
    private boolean concluido;
    private int prioridade;
    private long insercao;

    public Item(String nome){
        this.nome = nome;
        this.concluido = false;
        this.prioridade = 0;
    }

    public Item(String nome, int prioridade){
        this.nome = nome;
        this.concluido = false;
        this.prioridade = prioridade;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public boolean isConcluido() {
        return concluido;
    }

    public void setConcluido(boolean concluido) {
        this.concluido = concluido;
    }

    public int getPrioridade() {
        return prioridade;
    }

    public void setPrioridade(int prioridade) {
        this.prioridade = prioridade;
    }

    @Override
    public int compareTo(Item i) {
        int maiorPrioridade = Integer.compare(i.prioridade, this.prioridade);
        if (maiorPrioridade == 0) {
            // Se as prioridades forem iguais, ordena pela ordem de inserção (da mais antiga para a mais recente)
            return Long.compare(this.insercao, i.insercao);
        } else {
            return maiorPrioridade;
        }
    }
}
